models/articles.model.js

```diff
@@ -54,10 +60,34 @@ exports.readAllArticles = (sort_by = "created_at", order = "DESC") => {
         LEFT OUTER JOIN comments ON articles.article_id = comments.article_id 
         GROUP BY articles.author, articles.title, articles.article_id, articles.created_at, articles.topic, articles.votes, articles.article_img_url
         ORDER BY articles.${sort_by} ${upperCaseOrder};`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
+  } else {
+    await checkExists("topics", "slug", topic);
+
+    return db
+      .query(
+        `SELECT articles.author, 
+    articles.title, 
+    articles.article_id, 
+    articles.created_at, 
+    articles.topic, 
+    articles.votes, 
+    articles.article_img_url, 
+    CAST(COUNT(comments.comment_id) AS INTEGER) AS comment_count
+      FROM articles 
+      LEFT OUTER JOIN comments ON articles.article_id = comments.article_id 
+      WHERE articles.topic = $1
+      GROUP BY articles.author, articles.title, articles.article_id, articles.created_at, articles.topic, articles.votes, articles.article_img_url
+      ORDER BY articles.${sort_by} ${upperCaseOrder};`,
+        [topic]
+      )
+      .then(({ rows }) => {
+        return rows;
+      });
+  }
```

Comment:
Take a look at the notes for complex queries, you can build this string up in sections rather than doing a big if/else. It would tidy this model up quite a bit!